File: 4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.go
func findMedianSortedArrays(nums1 []int, nums2 []int) float64 {
    merged := []int{}
    i, j := 0, 0

    for i < len(nums1) && j < len(nums2) {
        if nums1[i] < nums2[j] {
            merged = append(merged, nums1[i])
            i++
        } else {
            merged = append(merged, nums2[j])
            j++
        }
    }
    merged = append(merged, nums1[i:]...)
    merged = append(merged, nums2[j:]...)

    totalSize := len(merged)
    if totalSize%2 == 0 {
        return float64(merged[totalSize/2-1]+merged[totalSize/2]) / 2.0
    }
    return float64(merged[totalSize/2])
}

File: 4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.py
from typing import List

class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        merged = []
        i, j = 0, 0

        while i < len(nums1) and j < len(nums2):
            if nums1[i] < nums2[j]:
                merged.append(nums1[i])
                i += 1
            else:
                merged.append(nums2[j])
                j += 1

        
        merged.extend(nums1[i:])
        merged.extend(nums2[j:])

        total_size = len(merged)
        if total_size % 2 == 0:
            return (merged[total_size // 2 - 1] + merged[total_size // 2]) / 2.0
        else:
            return float(merged[total_size // 2])
